core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -206,6 +207,14 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
             final EventLoop eventLoop = ctx.channel().eventLoop();
             req = DecodedHttpRequest.of(endOfStream, eventLoop, id, streamId, headers, true,
                                         inboundTrafficController, routingCtx);
+            if (Flags.allowLargeRequestEarlyRejection()) {
+                final DecodedHttpRequestWriter decodedReq =
+                        (req instanceof DecodedHttpRequestWriter) ? (DecodedHttpRequestWriter) req : null;
```

Comment:
optional; this logic could be simplified if `maxRequestLength` and `transferredBytes` are pulled down to `DecodedHttpRequest`

Reply:
good idea, done.